### PR TITLE
Updating visual studio references

### PR DESCRIPTION
**Pull request by jmarolf**
This change updates roslyn to use the released 15.3 binaries.  This is necessary for insertions to work.

Currently blocked on:
- [X] updating azure images to have latest 15.3 builds
- [ ] reconciling new JTF exceptions with TestWorkspace

### CODE REVIEW

**Comment by dpoeschl** [comment]
retest windows_release_vs-integration_prtest please

**Comment by dpoeschl** [comment]
@jmarolf The unit test failure looks pretty consistent between queues.

**Comment by jmarolf** [comment]
This is a real failure in composition that I'm working on

Thanks,
Jon
________________________________

File: src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs [hunk 1]
```diff
@@ -193,6 +194,7 @@ private static IList<Exception> Flatten(ICollection<Exception> exceptions)
             var aggregate = new AggregateException(exceptions);
             return aggregate.Flatten().InnerExceptions
                 .Select(UnwrapException)
+                .Where(ex => !(ex is JoinableTaskContextException))
```
Comment (jmarolf) [inline src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs]: New exceptions are now thrown if you attempt to access services on the UI thread

```
Microsoft.VisualStudio.Threading.JoinableTaskContextException: An attempt to switch to the main thread failed to reach the expected thread. Was the JoinableTaskContext initialized on the wrong thread or with a SynchronizationContext whose Post method does not execute its delegate on the main thread?
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.MainThreadAwaiter.GetResult()
   at Microsoft.VisualStudio.Text.Outlining.OutliningManager.<SourceTextChangedAsync>d__20.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.<>c__DisplayClass41_0`1.<<RaiseEventOnBackgroundAsync>b__0>d.MoveNext()
```

**Comment by dpoeschl** [comment]
VS test failures are being investigated.

**Comment by jmarolf** [comment]
retest windows_release_unit64_prtest please

File: src/VisualStudio/Core/Next/SymbolSearch/VisualStudioSymbolSearchService.ProgressService.cs [hunk 2]
```diff
@@ -72,7 +72,7 @@ private static TaskHandlerOptions GetOptions(string title)
             var options = new TaskHandlerOptions
             {
                 Title = title,
-                RetentionAfterCompletion = CompletionRetention.Faulted
+                ActionsAfterCompletion = CompletionActions.RetainOnFaulted
```
Comment (jmarolf) [inline src/VisualStudio/Core/Next/SymbolSearch/VisualStudioSymbolSearchService.ProgressService.cs]: @CyrusNajmabadi making this change to conform to the new api

Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Next/SymbolSearch/VisualStudioSymbolSearchService.ProgressService.cs]: Sure.

File: src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs [hunk 3]
```diff
@@ -66,16 +67,16 @@ public TestWorkspace(ExportProvider exportProvider, string workspaceKind = null,
         }
 
         /// <summary>
-        /// Reset the thread affinity, in particular the designated foreground thread, to the active 
-        /// thread.
```
Comment (jasonmalinowski) [inline src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs]: Although I don't mind that we're cleaning up the whitespace stuff here, can we make it a separate commit? Or do a big pass for the codebase first?

File: src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs (hunk 1 above)
Comment (jasonmalinowski) [inline src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs]: What is the right fix here?

Comment (jmarolf) [inline src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs]: The right fix here is to reconcile how we construct WPF text views with what JTF expects.  I'm currently undecided on whether this should block this PR.  If we care about inserting we may want to merge this and file a bug to resolve later.

Comment (rchande) [inline src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs]: @jmarolf You can just disable the outlining feature of the editor as I've done in a different PR. Our outlining unit tests directly test our outlining implementation without going through the editor.

https://github.com/dotnet/roslyn/pull/20397/files#diff-6159a3d84595fb60c93e257f29db681d

Comment (jmarolf) [inline src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs]: If we don't care about this case, or its going away in @rchande 's PR then I suggest we merge this once the azure VMs are updated so we can begin inserting again.  We can remove this once @rchande 's changes make it to master

**Comment by jmarolf** [comment]
@jasonmalinowski @dotnet/roslyn-ide Any objections to merging this?

**Comment by jmarolf** [comment]
files Issue https://github.com/dotnet/roslyn/issues/21245

**Comment by jaredpar** [comment]
> This is necessary for insertions to work.

Why?
